gto-support-core/src/main/java/org/ccci/gto/android/common/util/AsyncTaskCompat.java

```diff
@@ -11,13 +11,37 @@
 import java.util.concurrent.TimeUnit;
 
 public class AsyncTaskCompat {
-    private static final Object LOCK_EXECUTOR = new Object();
-    private static Executor EXECUTOR;
+    private static final Compat COMPAT;
+    static {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
+            COMPAT = new FroyoCompat();
+        } else {
+            COMPAT = new HoneycombCompat();
+        }
+    }
+
+    public static void execute(@NonNull final Runnable task) {
+        COMPAT.execute(task);
+    }
 
     @NonNull
-    @TargetApi(Build.VERSION_CODES.GINGERBREAD)
-    private static Executor getExecutor() {
-        synchronized (LOCK_EXECUTOR) {
+    public static Executor SERIAL_EXECUTOR() {
```

Comment:
I'm not a huge fan of that syntax either. I was torn because for the Compat classes I try to keep the same names as the corresponding method in the Android API. This is the first time I've made a compat mechanism for a constant. And to further complicate that, this is a Constant pointing to an executor, which isn't lightweight. So I wanted to try and be lazy about initializing it.

I just had an idea for a potential better solution, let me try that out and see if it's better.